### PR TITLE
Push to docker hub

### PR DESCRIPTION
**Pull request by benjamin-bergia**
Run the build and tests only on PR
Push to docker hub only during push on master

### CODE REVIEW

**Review by drjova — APPROVED** [review]

File: .github/workflows/push-to-docker-hub.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: Retrieve cached image
+        uses: actions/cache@v2
+        env:
+          cache-name: docker-images
+        with:
+          path: ${{ env.IMAGE_FILE }}
+          key: inspirehep-hep-${{ hashFiles('backend/poetry.lock') }}-${{ github.sha }}
```
Comment (michamos) [inline .github/workflows/push-to-docker-hub.yml:25]: are you sure this works? The sha might change between the PR and the commit on master.

Comment (benjamin-bergia) [reply .github/workflows/push-to-docker-hub.yml:25]: Currently we rebase only. Does it change on rebase too?

Comment (michamos) [reply .github/workflows/push-to-docker-hub.yml:25]: yes, in general it does

**Comment by benjamin-bergia** [comment]
So after our discussion @michamos on Friday, I rewrote this workflow to use the docker registry and use the pull_request_target. I don't really see how this could be tested. Maybe if somebody wants to fork my fork and open a PR on my fork? Or we can just merge this and see.

**Review by michamos — APPROVED** [review]

**Pull request closed without merging**